ydb_dbapi/connection.py

```diff
@@ -0,0 +1,217 @@
+from typing import (
+    Any,
+    List,
+    NamedTuple,
+    Optional,
+)
+import posixpath
+
+import ydb
+from ydb.retries import retry_operation_async
+
+from .cursors import Cursor
+
+from .utils import (
+    handle_ydb_errors,
+)
+
+from .errors import (
+    InterfaceError,
+    InternalError,
+    NotSupportedError,
+)
+
+
+class IsolationLevel:
+    SERIALIZABLE = "SERIALIZABLE"
+    ONLINE_READONLY = "ONLINE READONLY"
+    ONLINE_READONLY_INCONSISTENT = "ONLINE READONLY INCONSISTENT"
+    STALE_READONLY = "STALE READONLY"
+    SNAPSHOT_READONLY = "SNAPSHOT READONLY"
+    AUTOCOMMIT = "AUTOCOMMIT"
+
+
+class Connection:
+    def __init__(
+        self,
+        host: str = "",
+        port: str = "",
+        database: str = "",
+        **conn_kwargs: Any,
```

Comment:
**suggestion (blocking)**: Can we use TypedDict here? And I think we need to add one more argument `timeout` to configure gRPC call timeouts. Now each call DDL or DML or even Describe will wait 60 seconds before terminate. It needs to be configured.